api/question/schema_query.py

```diff
@@ -29,12 +34,32 @@ def get_room_response(self, unique_responses, question_id, question_type):
         response = []
 
         for room in unique_responses:
-            room_name = Room.query.filter_by(id=room.room_id).first().name   # noqa: E501
-            responses_in_room = ResponseModel.query.filter_by(room_id=room.room_id, question_id=question_id)  # noqa: E501
-            response_count = ResponseModel.query.filter_by(room_id=room.room_id).count()  # noqa: E501
-            if question_type == "rate":
-                average_rating = sum([response.rate for response in responses_in_room]) / responses_in_room.count()  # noqa: E501
-                room_response = ResponsePerRoom(room_name=room_name, response_count=response_count, cleanliness_rating=average_rating)  # noqa: E501
+            room_name = Room.query.filter_by(id=room.room_id).first().name
+            responses_in_room = ResponseModel.query.filter_by(
+                room_id=room.room_id, question_id=question_id)
+            response_count = ResponseModel.query.filter_by(
+                room_id=room.room_id).count()
+            average_rating = (sum(
+                [response.rate for response in responses_in_room]) /
+                 responses_in_room.count())
+            all_resources = ResourceModel.query.filter_by(
+                room_id=room.room_id).count()
+            all_missing_resources = ResourceModel.query.filter(
+                ResourceModel.resources.any(room_id=room.room_id)).count()
+            total_missing_resources = '{} out of {}'.format(
+                all_missing_resources, all_resources)
+            comment = ResponseModel.query.filter_by(
+                room_id=room.room_id).filter(column("additional_comment").isnot(None))  # noqa: E501
+            if comment.count() > 0:
+                last_additional_comment = comment[-1].additional_comment
+            else:
+                last_additional_comment = "No comment yet"
```

Comment:
What if as a user I give my response in the comment box as "No comment yet", how will we differentiate my comment from this response. Perhaps you could give a more descriptive response to differentiate my comment as a user from this API response.

Reply:
Noted @Bonifase, I'll get to that immediately.